relay-common/src/glob3.rs

```diff
@@ -12,6 +12,43 @@ fn is_match(globs: &[Regex], message: &[u8]) -> bool {
     globs.iter().any(|regex| regex.is_match(message.as_ref()))
 }
 
+/// A singular glob-pattern.
+#[derive(Debug, Clone, PartialEq, Default)]
+pub struct GlobPattern(GlobPatterns);
```

Comment:
I would rather redefine `GlobPatterns` as a wrapper around `Vec<GlobPattern>`. That would save you the custom serialization. Question is, why not use `glob2::LazyGlob`?

Reply:
`GlobPatterns` is an old type though, and when it comes to multiple patterns it does what we want it to do. How about keeping it as-is and just getting rid of `GlobPattern` and replace with `LazyGlob` as you suggested? i'm pushing those changes now